tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/multiprotocol/JMSCorrelationIDTest.java

```diff
@@ -0,0 +1,186 @@
+
+      Message m = consumer.receive(5000);
+      Assert.assertNotNull("Could not receive message on consumer", m);
+
+      Assert.assertEquals(correlationId, m.getJMSCorrelationID());
+   }
+
+   @Test(timeout = 60000)
+   public void testCorrelationIDAsStringSendReceiveFromAMQPToAMQP() throws Throwable {
+      testCorrelationIDAsStringSendReceive(createConnection(), createConnection());
+   }
+
+   @Test(timeout = 60000)
+   public void testCorrelationIDAsStringSendReceiveFromAMQPToCore() throws Throwable {
+      testCorrelationIDAsStringSendReceive(createConnection(), createCoreConnection());
+   }
+
+   @Test(timeout = 60000)
+   public void testCorrelationIDAsStringSendReceiveFromAMQPToOpenWire() throws Throwable {
+      testCorrelationIDAsStringSendReceive(createConnection(), createOpenWireConnection());
+   }
+
+   @Test(timeout = 60000)
+   public void testCorrelationIDAsStringSendReceiveFromCoreToCore() throws Throwable {
+      testCorrelationIDAsStringSendReceive(createCoreConnection(), createCoreConnection());
+   }
+
+   @Test(timeout = 60000)
+   public void testCorrelationIDAsStringSendReceiveFromCoreToAMQP() throws Throwable {
+      testCorrelationIDAsStringSendReceive(createCoreConnection(), createConnection());
+   }
+
+   @Test(timeout = 60000)
+   public void testCorrelationIDAsStringSendReceiveFromCoreToOpenWire() throws Throwable {
+      testCorrelationIDAsStringSendReceive(createCoreConnection(), createOpenWireConnection());
+   }
+
+   @Test(timeout = 60000)
+   public void testCorrelationIDAsStringSendReceiveFromOpenWireToOpenWire() throws Throwable {
+      testCorrelationIDAsStringSendReceive(createOpenWireConnection(), createOpenWireConnection());
+   }
+
+   /*
+    * JMS supports setting the correlation ID as a String or a byte[]. However, OpenWire only supports correlation ID as
+    * a String. When it is set as a byte[] the OpenWire JMS client just converts it to a UTF-8 encoded String, and
+    * therefore when it sends a JMS message with a correlation ID the broker can't tell if the value was set as a String
+    * or a byte[]. Due to this ambiguity the broker is hard-coded to treat the value as a byte[]. This doesn't cause any
+    * problems if the consumer is also OpenWire, but if the consumer is core or AMQP (which both differentiate between
+    * String and binary values) then retrieving the correlation ID as a String (i.e. via Message.getJMSCorrelationID())
+    * will fail.
+    *
+    * JMS means for the correlation ID as a byte[] to be used for "native" clients which makes it a good candidate for
+    * interoperability between other protocols like MQTT 5 which *only* supports correlation ID as byte[].
+    */
+   @Ignore
+   @Test(timeout = 60000)
+   public void testCorrelationIDAsStringSendReceiveFromOpenWireToAMQP() throws Throwable {
+      testCorrelationIDAsStringSendReceive(createOpenWireConnection(), createConnection());
+   }
```

Comment:
Seems that you actually hit the interop issues I commented on from looking at the code.

Why is the broker 'hard coded to byte[]' for Openwire when this comment explicitly notes it effectively only does String? Why isnt it hard coded to using String...like it was before?

If MQTT only supports byte[] then it seem like it is the MQTT stuff that should be jumping through hoops such as converting to/from a UTF-8 bytes, not the Openwire bits, especially as doing it this way breaks the typical+existing Openwire<->AMQP/Core interop.